### DISCUSSION TITLE
How to setup the hardware properly?

### DISCUSSION THREAD

**Discussion by lvmasterrj** [vicwomg/pikaraoke#222] [Q&A]
Hello everyone.

I already have pikaraoke installed and running flawlessly, but I yet need a microphone system.

I've seen @vicwomg suggestion on the amazon.com microphone system, but that got me thinking on how to use only the rPi, the microphone system and the TV (no amplifiers or external speakers).

One idea I had was using some P3 to RCA cables, that you can find on the internet, is that the better option?! The yellow RCA (video cable) goes directly to the TV while the white an red goes on the input of the microfone system and then the output of the system goes to the TV input.

Another option that I don't know if would work, is to use the HDMI for the video and a P2 to RCA for the audio.

Also, searching on AliExpress, I got into [this microphone mixer](https://aliexpress.com/item/1005003536494747.html), that I think is a very good option for using with the pikaraoke, what you think?!

Sorry for some english flaws and thanks for the help.

**Comment by Hartebee5t** [comment]
The reason for the suggestion in the Readme to route the audio mix to a separate sound system is because the Pi doesn't provide a decent audio-in method (without a suitable HAT). 
My own PiKaraoke set-up uses a wireless mic mixer similar to the Amazon one suggested in @vicwomg's Readme notes. Mine has two UHF wireless mics and a Bluetooth input channel. The one you've found has more mixer features than mine but the Aliexpress description doesn't say what radio frequency band the wireless mic link uses. They should really be UHF for the audio quality (or VHF if it's all there is) but definitely NOT the Bluetooth input channel into the mixer. Bluetooth adds far too much delay into the live vocal to be usable. I'm not saying that the product is not suitable, merely that you should check with the seller whether the mics are Bluetooth.

By contrast, though, that Bluetooth *input* channel is great for getting the backing-music audio from the Pi to the mixer. It's how I run my own system. Even a half-second mismatch between the lyrics changing colour on the screen vs. the exact beat in the backing music that you're singing along with is not a noticeable delay, but both your vocal track input and the mixed output must be as 'live' as possible, otherwise you'll be singing against a horrible "slap-back echo" and, believe me, it's impossible to sing ahead of the music to compensate. That means you can't use a Bluetooth *output* link from the mixer to the audio amplifier either (direct analog connection is best). 

For the amplifier a cheap (older technology) analog soundbar, or a home hi-fi system with decent speakers works well in a home set-up. There are also plenty of guitar combo amps or those "party boom-boxes" around which would work well too. Avoid modern digital soundbars though. Even if you can find one with an "AUX" input (e.g. red/white RCA jacks) the processing will most likely add a noticeable delay to any direct analog audio connection from the mixer. The same is likely to be true for many modern TVs now too. They're not designed with a truly live performance in mind.

I hope you can find something to suit your needs (and budget).  Let the community know on this thread how you get on. I'd be interested in your review of that Aliexpress mic mixer/receiver if you go for that one.

Regards
David (@Hartebeest)

**Reply by lvmasterrj** [reply]
Thanks for your thoughts David.

I'm prone to getting this Aliexpress mixer, pehaps I'll wait the end of the hollydays.

The description of the mics says "Cha 576.1-590.1mhz, Chb 560.1-574.1mhz" so, i gess UHF it is. The bluetooth seems to be only to receive music/sound from a device, not to transmit.

I thought very interesting the thing you've said about modern digital devices on processing the audio thus adding a delay to it. I'll never thought about that, now I'll be prepared to understand if that happens.

One thing that got me thinking is how you manage to output the audio of the pikaraoke by bluetooth to the mixer, and also, how exactly is the setup (HDMI on TV only for image and bluetooth audio to the mixer and out to a speaker, pehaps?)

Thanks again.

**Reply by Hartebee5t** [reply]
Ah, how to get Bluetooth audio out of the Pi? I wish I could remember how I got that working...
I do need to write that up. I think there's an issue posted by someone else in this repository somewhere. I'll see if I can find it (or remember how myself).
I *will* get back to you about this, but it might take a few weeks.

**Reply by lvmasterrj** [reply]
No problem.

Now I know that's a possibility.

Thanks anyway.
